media/js/helpfulness.js

```diff
@@ -0,0 +1,91 @@
+(function(win, doc, $) {
+
+    // this feature requires localStorage
+    if (('localStorage' in win)) {
+        var ignore = localStorage.getItem('helpful-ignore') === 'true'; // true if ever clicked ignore
+        var asked_recently = parseInt(localStorage.getItem(doc.location + '#answered-helpful'), 10) > Date.now();
+        if (!ignore && !asked_recently) {
+            // ask about helpfulness after 1 min
+            setTimeout(inquire, 60000);
+        }
+    }
+
+    // create a notification
+    function inquire() {
+        // dimension7 is "helpfulness"
+        ga('set', 'dimension7', 'Yes');
```

Comment:
I think I understand why this is a good idea in principle. But we're setting this dimension flag for users flagged by waffle, who have not said "Never ask me again" and who have not already responded to the inquiry on the page they're looking at. Replicating all that non-GA business logic in the GA file seems like a violation of DRY. Do you see another way to do it, or do you think maybe this should be an exception to the rule?